src/examples/amalgamation_header.cpp

```diff
@@ -0,0 +1,15 @@
+#include <iostream>
+#if __has_include(<botan/botan_all.h>)
+   #include <botan/botan_all.h>
```

Comment:
```suggestion
#if __has_include(<botan/botan_all.h>)
   #include <botan/botan_all.h>
#else
   // if the amalgamation header isn't available, you have to IWYU.
   #include <botan/hex.h>
#endif
```

... no need for the fallback main function then.